src/storage/v2/indices/text_index.cpp

```diff
@@ -409,21 +420,15 @@ std::string TextIndex::Aggregate(const std::string &index_name, const std::strin
 }
 
 void TextIndex::Commit() {
-  if (!flags::AreExperimentsEnabled(flags::Experiments::TEXT_SEARCH)) {
-    throw query::TextSearchDisabledException();
-  }
-
   for (auto &[_, index_data] : index_) {
+    std::lock_guard lock(index_writer_muex_);
```

Comment:
Check whether we need to take lock outside or inside the loop. Same for rollback

Reply:
We probably don't need Commit and CommitLoadedNodes, check if do twice the work